scripts/analysis.py

```diff
@@ -638,50 +583,171 @@ def final_stockpile(cursor, facility):
+                          'INNER JOIN agententry ON '
+                          'transactions.receiverid = agententry.agentid '
+                          'WHERE spec LIKE "%Reactor%" '
+                          'GROUP BY time').fetchall()
+
+    return get_timeseries(fuel, duration, .001, 'TRUE')
+
+
+def conv_ratio(cursor, in_, out, is_recipe):
+    """ Returns conversion ratio of two commodities
 
     Parameters
     ----------
     cursor: sqlite cursor
         sqlite cursor
-    fuel_list: list
-        list of fuel commodity names (eg. uox, mox) to consider in fuel usage.
+    in_: str
+        name of in recipe / commodity
+    out: str
+        name of out recipe / commodity
+    is_recipe: bool
+        recipe if true, commodity if false
 
     Returns
     -------
-    dict
-        dictionary of different fuels used timeseries
-    keys - fuel
-    values - timeseries list of fuel amount [kg]
+    prints conversion ratio
     """
 
-    cur = cursor
-    fuel_dict = collections.OrderedDict()
-    for fuel in fuel_list:
-        temp_list = ['"' + fuel + '"']
-        fuel_quantity = cur.execute(exec_string(temp_list, 'commodity',
-                                    'time, sum(quantity)')
-                                    + ' GROUP BY time').fetchall()
-        init_year, init_month, duration, timestep = get_sim_time_duration(cur)
-        total_sum = 0
-        quantity_timeseries = []
+    if is_recipe:
+        in_qualid = cursor.execute('SELECT qualid FROM recipes '
+                                   'WHERE recipe = "' + in_ + '"').fetchone()[0]
+        out_qualid = cursor.execute('SELECT qualid FROM recipes '
+                                    'WHERE recipe = "' + out + '"').fetchone()[0]
+    else:
+        in_qualid = cursor.execute(exec_string(['"' + in_ + '"'],
+                                                      'commodity', 'qualid')).fetchone()[0]
+        out_qualid = cursor.execute(exec_string(['"' + out + '"'],
+                                                       'commodity', 'qualid')).fetchone()[0]
+    in_recipe = cursor.execute('SELECT nucid, massfrac FROM compositions '
+                               'WHERE qualid = ' + str(in_qualid)).fetchall()
+    out_recipe = cursor.execute('SELECT nucid, massfrac FROM compositions '
+                                'WHERE qualid = ' + str(out_qualid)).fetchall()
+
+    fissile_list = [922350000, 942410000, 942390000]
+    FP = sum([massfrac for (nucid, massfrac)
+              in out_recipe if 350000000 < nucid < 890000000])
+    fissile_in_spent = sum(
+        [massfrac for (nucid, massfrac) in out_recipe if nucid in fissile_list])
+    in_fissile = sum([massfrac for (nucid, massfrac)
+                      in in_recipe if nucid in fissile_list])
+    print('The Conversion Factor is:')
+    print((FP + fissile_in_spent - in_fissile) / FP)
+
+
+def mix_ratio(cursor, fuel_recipe_name, spent_fuel_recipe_name, depleted_u_recipe_name, what_reprocess):
```

Comment:
I pulled out the block where it finds the optimal ratio into a separate function!